src/slack_planning_bot/core/spec.clj

```diff
@@ -0,0 +1,65 @@
+(ns slack-planning-bot.core.spec
+  (require [slack-planning-bot.core.entity :as entity]
+           [clojure.spec.alpha :as s]))
+
+(s/def ::user-id string?)
+(s/def ::message string?)
+
+(s/def ::messenger #(satisfies? entity/Messenger %))
+(s/def ::messenger-error keyword?)
+(s/def ::messenger-result (s/or :data (s/map-of ::user-id ::message)
+                            :error ::messenger-error))
+
+(s/def ::time-tracker #(satisfies? entity/TimeTracker %))
+(s/def ::time-tracker-error keyword?)
+(s/def ::time-tracker-result (s/or :data () :error ::time-tracker-error))
```

Comment:
Why `()`?

Reply:
oh this one is empty because I didn't know what it was returning